pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -1522,21 +1522,18 @@ protected void internalGetPartitionedStats(AsyncResponse asyncResponse, boolean
                     }
                 }
                 if (perPartition && stats.partitions.isEmpty()) {
-                    try {
-                        boolean pathExists = namespaceResources().getPartitionedTopicResources()
-                                .partitionedTopicExists(topicName);
-                        if (pathExists) {
-                            stats.partitions.put(topicName.toString(), new TopicStatsImpl());
-                        } else {
-                            asyncResponse.resume(
-                                    new RestException(Status.NOT_FOUND,
-                                            "Internal topics have not been generated yet"));
-                            return null;
-                        }
-                    } catch (Exception e) {
-                        asyncResponse.resume(new RestException(e));
-                        return null;
-                    }
+                    return namespaceResources().getPartitionedTopicResources()
+                            .partitionedTopicExistsAsync(topicName).
+                            thenAccept(exists -> {
+                                if (exists) {
+                                    stats.partitions.put(topicName.toString(), new TopicStatsImpl());
+                                    asyncResponse.resume(stats);
+                                } else {
+                                    asyncResponse.resume(
+                                            new RestException(Status.NOT_FOUND,
+                                                    "Internal topics have not been generated yet"));
+                                }
+                            });
```

Comment:
it seems that `return ` should be omitted from line 1525 and `return null;` should be before the block ends.

Reply:
@lhotari Lines 1530 and 1538 are written the stats data to the HTTP client. 

Do you happen to miss line 1530?

Reply:
I meant a different thing. The return on line 1525 should be removed and instead there should be a `return null;` after that expression.

Reply:
@lhotari Done.